astropy/coordinates/baseframe.py

```diff
@@ -276,6 +276,68 @@ def convert_input(self, value):
         return out, converted
 
 
+class CartesianRepresentationFrameAttribute(FrameAttribute):
+    """
+    A frame attribute that is a CartesianRepresentation with specified units.
+
+    Parameters
+    ----------
+    default : object
+        Default value for the attribute if not provided
+    secondary_attribute : str
+        Name of a secondary instance attribute which supplies the value if
+        ``default is None`` and no value was supplied during initialization.
+    unit : unit object or None
+        Name of a unit that the input will be converted into. If None, no
+        unit-checking or conversion is performed
+    """
+    def __init__(self, default=None, secondary_attribute='', unit=None):
+        super(CartesianRepresentationFrameAttribute, self).__init__(default, secondary_attribute)
+        self.unit = unit
+
+    def convert_input(self, value):
+        """
+        Checks that the input is a CartesianRepresentation with the correct unit, or
+        the special value ``[0, 0, 0]``.
+
+        Parameters
+        ----------
+        value : object
+            Input value to be converted.
+
+        Returns
+        -------
+        out, converted : correctly-typed object, boolean
+            Tuple consisting of the correctly-typed object and a boolean which
+            indicates if conversion was actually performed.
+
+        Raises
+        ------
+        ValueError
+            If the input is not valid for this attribute.
+        """
+        if (isinstance(value, list) and np.all([v == 0 for v in value]) and
```

Comment:
I don't think that works since if ```value``` is a CartesianRepresentation it will raise an exception (they cannot be indexed). 

Let me think about it.